### PR TITLE
feat: add fmt.Stringer interface to all result types

### LINKED ISSUE #20: Add fmt.Stringer interface to result types
## Priority: P3

Result types `SunEvent`, `MoonEvent`, `Transit`, `TwilightEvent`, `LunarPhaseInfo`, `Equatorial`, `Horizontal`, and `Ecliptic` lack `fmt.Stringer` implementations.

Adding `String()` methods would improve debugging and logging ergonomics for library consumers.

Found during comprehensive v2 review.

### PR DESCRIPTION
**Pull request by philoserf**
## Summary

- Add `String()` methods to all 8 exported result types for debugging and logging ergonomics
- `Equatorial`, `Horizontal`, `Ecliptic` show coordinates with 3 decimal places
- `SunEvent`, `MoonEvent`, `Transit`, `TwilightEvent` show times as `15:04` (or `--:--` for zero values) with duration
- `LunarPhaseInfo` shows phase name, illumination percentage, and day into lunation

## Changes

- **`stringer.go`** (new): `String()` methods + `formatTime` helper
- **`stringer_test.go`** (new): tests for all types including zero-value edge case

## Testing

- All new and existing tests pass with `-race`
- Full quality gate clean: `go vet`, `golangci-lint`, `gofmt`

## Related Issues

Closes #20

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: stringer.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+func (m MoonEvent) String() string {
+	return fmt.Sprintf("Rise=%s Set=%s Duration=%s",
+		formatTime(m.Rise),
+		formatTime(m.Set),
+		m.Duration)
+}
+
+// String returns a human-readable representation of the lunar phase.
+func (l LunarPhaseInfo) String() string {
+	return fmt.Sprintf("%s %.1f%% (day %.1f)", l.Name, l.Illumination, l.DaysApprox)
```
[the remark this answers is not in the document]
Comment (philoserf) [reply stringer.go]: Addressed in c0ef2ea — adopted the suggested namePart approach.
